godal.go

```diff
@@ -437,6 +437,74 @@ func (band Band) Histogram(opts ...HistogramOption) (Histogram, error) {
 	return h, nil
 }
 
+//Statistics returns if present and flag as true.
```

Comment:
The comment and signature for this function need to be fixed:

- comment should start with `GetStatistics`
- comment should mention that it will only return *cached* statistics and will not compute them
- returns false and *no* error if no statistics are available
- comment should list available options

By convention, the return order should be `(Statistics, bool, error)` instead of `(bool, Statistics, error)`